flatliners/prometheusendpoint.py

```diff
@@ -0,0 +1,66 @@
+from time import sleep
+from collections import defaultdict
+from .baseflatliner import BaseFlatliner
+
+from prometheus_client import Gauge, start_http_server
+
+# Set up logging
+import logging
+_LOGGER = logging.getLogger(__name__)
+
+class PrometheusEndpoint(BaseFlatliner):
+    """docstring for PrometheusEndpoint."""
+    def __init__(self):
+        super().__init__()
+        _LOGGER.info("Prometheus Endpoint initialized.")
+        self.update_interval = 60 # seconds
+        self.metric_buffer = defaultdict(list)
+        self.weirdness_score_gauge = Gauge('weirdness_score','Weirdness score for the given Cluster and Version',['cluster','version'])
+        self.prev_metric_list = []
+
+    def on_next(self, x):
+        try:
+            rounded_timestamp = round(float(x.timestamp)/self.update_interval) * self.update_interval
+        except Exception as e:
+            _LOGGER.error("Couldn't process the following packet {0}. Reason: {1}".format(x,str(e)))
+            raise e
+        self.metric_buffer[rounded_timestamp].append([x.cluster, x.version, x.weirdness_score])
+        pass
+
+    def _update_exposed_metrics(self):
+        if self.metric_buffer:
+            earliest_timestamp = min(self.metric_buffer)
+            for metric_values in self.metric_buffer[earliest_timestamp]:
+                self.weirdness_score_gauge.labels(cluster=str(metric_values[0]), version=str(metric_values[1])).set(metric_values[2])
```

Comment:
Could you explain why you do the buffering and the update_interval?
Why not just do this call `self.weirdness_score_gauge.labels().set()` in the `on_next` method?

Reply:
Because if we do it without a buffer, for each cluster and version pair we would expose only one value per chunk of data processed. 
So for example, if we have the chunk size set to 1 hour, then for each cluster and version pair we would only expose 1 value every hour. This way the exposed value changes as frequently as the incoming data. (Testing on some real data says that happens about every 3-5 minutes)